src/modm/architecture/interface/interrupt.hpp

```diff
@@ -110,6 +111,7 @@
 #	define MODM_ISR_CALL(vector) \
 		vector ## _IRQHandler()
 #	define MODM_ISR(vector, ...) \
+		static_assert(!std::is_void<decltype(vector ## _IRQn)>::value, "No such IRQ " #vector); \
```

Comment:
You probably want to `#ifdef __cplusplus` this and the header include, since these macro are also callable from a pure C context.

Reply:
I would prefer some kind of C version of this check, as C17 supports `static_assert` via `<assert.h>`. We can include a platform specific header file that contains a list of some sort with the generated vector table names, so that's easy, I just don't know how to check the membership of a string (?) in a set of strings in C at compile time.

Reply:
Ha, perhaps we can even generate the static_assert message? That would be cool "This is not a valid IRQ, but here's a list of all valid IRQs". We do this for [the STM32F1 remap groups](https://modm.io/how-modm-works/#modm-asserts-at-compile-time).

Reply:
Hmm...OK good point. It's pretty easy to use the C assert, and that's probably what we should do. But that makes me a little sad, because I just learned about the `string_view_literal` and something like this with a python generated list of valid IRQ name strings would let us actually fail the assert and print whatever we want in C++. 

```
#include <string_view>

using namespace std::string_view_literals;
constexpr std::array valid_strings = {"Word"sv, "list"sv};

constexpr bool valid_word(const std::string_view &s) {
    for(const auto &valid : valid_strings) {
        if(valid == s) return true;
    }
    return false;
}
static_assert(valid_word("Word"), "Why'd this fail?");
static_assert(valid_word("ABC"), "This should fail!");
```

Sadly, I've no idea how to do it in C.

Reply:
That's ok, we can have different implementations for C and C++ via `#ifdef`. It's ok if we don't immediately have a C version, the C++ version is very, very good, you should definitely implement that!